File: sqlparser/src/sql/csgen/SqlValueExpression.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
using System.Globalization;
namespace CData.Sql {

public sealed class SqlValueExpression : SqlExpression {
  private SqlValue value;
  private readonly string paraName;

  public SqlValueExpression(SqlValueType valueType, string v) : this(null, new SqlValue(valueType, v)) {
  }

  public SqlValueExpression(SqlValueType valueType, string v, CultureInfo culture) : this(null, new SqlValue(valueType, v, culture)) {
  }

  public SqlValueExpression(string paraName) : this(paraName, null) {
  }

  public SqlValueExpression(SqlValue value) : this(null, value) {
  }

  public SqlValueExpression(string paraName, SqlValue paraValue) {
    this.value = paraValue;
    this.paraName = paraName;
  }

  public override SqlValue Evaluate() {
    if (value == null) {
      return SqlValue.GetNullValueInstance();
    }
    return value;
  }

  public override bool IsEvaluatable() {
    return value != null;
  }

  public bool IsParameter() {
    return this.paraName != null && !this.paraName.Equals("");
  }

  public string GetParameterName() {
    return this.paraName;
  }

  public Object GetParameterValue() {
    if (this.value != null) {
      return this.value.GetOriginalValue();
    } else {
      return null;
    }
  }

  public string GetParameterValueAsString() {
    if (this.value != null) {
      return this.value.GetValueAsString(null);
    } else {
      return null;
    }
  }

  public int GetParameterDataType() {
    if (this.value != null) {
      return this.value.GetDataType();
    } else {
      return ColumnInfo.DATA_TYPE_NOT_SPECIFIED;
    }
  }

  public string GetValueAsNamedParameter(RebuildOptions rebuildOptions){
    string pattern = Utilities.GetValueAsString(rebuildOptions.GetParaNamePattern(), RebuildOptions.DEFAULT_PARA_NAME_PATTERN);
    string value = "";
    if (this.paraName != null && this.paraName.Length > 0) {
      value = RSSBus.core.j2cs.Converter.GetSubstring(paraName, 1);
    }
    return Utilities.FormatString(pattern, new string [] {value});
  }

  public void UpdateValue(SqlValue value) {
    this.value = value;
  }

  public override Object Clone() {
    SqlValueExpression obj = new SqlValueExpression(this.paraName, null);
    obj.Copy(this);
    return obj;
  }

  protected override void Copy(SqlExpression obj) {
    SqlValueExpression o = (SqlValueExpression)obj;
    this.value = o.value == null ? null : (SqlValue) o.value.Clone();
  }
}
}

File: sqlparser/src/sql/SqlValueExpression.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
using System.Globalization;
namespace CData.Sql {
#*/
import rssbus.oputils.common.Utilities;
import java.util.Locale;

public final class SqlValueExpression extends SqlExpression {
  private SqlValue value;
  private final String paraName;

  public SqlValueExpression(SqlValueType valueType, String v) {
    this(null, new SqlValue(valueType, v));
  }

  public SqlValueExpression(SqlValueType valueType, String v, /*#CultureInfo#*//*@*/Locale/*@*/ culture) {
    this(null, new SqlValue(valueType, v, culture));
  }

  public SqlValueExpression(String paraName) {
    this(paraName, null);
  }

  public SqlValueExpression(SqlValue value) {
    this(null, value);
  }

  public SqlValueExpression(String paraName, SqlValue paraValue) {
    this.value = paraValue;
    this.paraName = paraName;
  }

  public /*#override#*/ SqlValue evaluate() {
    if (value == null) {
      return SqlValue.getNullValueInstance();
    }
    return value;
  }

  public /*#override#*/ boolean isEvaluatable() {
    return value != null;
  }

  public boolean isParameter() {
    return this.paraName != null && !this.paraName.equals("");
  }

  public String getParameterName() {
    return this.paraName;
  }

  public Object getParameterValue() {
    if (this.value != null) {
      return this.value.getOriginalValue();
    } else {
      return null;
    }
  }

  public String getParameterValueAsString() {
    if (this.value != null) {
      return this.value.getValueAsString(null);
    } else {
      return null;
    }
  }

  public int getParameterDataType() {
    if (this.value != null) {
      return this.value.getDataType();
    } else {
      return ColumnInfo.DATA_TYPE_NOT_SPECIFIED;
    }
  }

  public String getValueAsNamedParameter(RebuildOptions rebuildOptions){
    String pattern = Utilities.getValueAsString(rebuildOptions.getParaNamePattern(), RebuildOptions.DEFAULT_PARA_NAME_PATTERN);
    String value = "";
    if (this.paraName != null && this.paraName.length() > 0) {
      value = paraName.substring(1);
    }
    return Utilities.formatString(pattern, new String [] {value});
  }

  public void updateValue(SqlValue value) {
    this.value = value;
  }

  @Override
  public Object clone() {
    SqlValueExpression obj = new SqlValueExpression(this.paraName, null);
    obj.copy(this);
    return obj;
  }

  @Override
  protected void copy(SqlExpression obj) {
    SqlValueExpression o = (SqlValueExpression)obj;
    this.value = o.value == null ? null : (SqlValue) o.value.clone();
  }
}
